### PR TITLE
Trigger events on collection operations (ref #490)

### PR DESCRIPTION
**Pull request by leplatrem**
* [x] Send events
* [x] Adjust payload
* [x] Should we have a single event `"changed" {action: "create", ...} ` ?
* [x] feedback? @glasserc 
* [x] Documentation

### CODE REVIEW

**Comment by glasserc** [comment]
How should this interact with transactions? My current implementation of this runs a transaction with a bunch of sets (or deletes, or whatever) and then emits one event only if the transaction succeeds. Of course, this means that the format of the event is like `{id: {oldValue: ..., newValue: ...}}`.

**Comment by leplatrem** [comment]
> then emits one event only if the transaction succeeds

Good catch! I'll rework this!

File: docs/api.md [hunk 1]
```diff
@@ -765,6 +767,60 @@ kinto.events.on("deprecated", function(event) {
...
+The `event` argument contains the following information:
+
+- `targets`: a list of mappings with an `action` attribute and the other attributes of the respective event as described below.
+
+```js
+const kinto = new Kinto();
+const articles = kinto.collection("articles");
+
+articles.events.on("change", function(event) {
+  const first = events.targets[0];
```
Comment (glasserc) [inline docs/api.md]: I think this should be `event`.

File: docs/api.md [hunk 2]
```diff
@@ -765,6 +767,60 @@ kinto.events.on("deprecated", function(event) {
...
+
+#### Per action event
+
+Atomic events are sent for ``create``, ``update`` and ``delete``.
+
+Every event contains the following attributes:
+
+- ``data``: the record that was created, updated or deleted
+
+The ``update`` event contains an additionnal attribute:
```
Comment (glasserc) [inline docs/api.md]: additional

Comment (glasserc) [inline docs/api.md]: Also, is there an `id` provided?

File: src/collection.js [hunk 3]
```diff
@@ -1159,6 +1161,27 @@ export class CollectionTransaction {
   constructor(collection, adapterTransaction) {
     this.collection = collection;
     this.adapterTransaction = adapterTransaction;
+
+    this._events = [];
+  }
+
+  _windEvent(action, payload) {
```
Comment (glasserc) [inline src/collection.js]: I've never seen the word "wind"/"wound" used in this kind of context. It's only a private method so it's probably not a big deal. What do you think about "queued" or "buffered"?

File: test/collection_test.js [hunk 4]
```diff
@@ -2447,4 +2447,160 @@ describe("Collection", () => {
...
+    beforeEach(() => {
+      articles = testCollection();
+      return articles.create({title: "foo"})
+        .then(({data}) => article = data);
+    });
+
+    it("should emit an event on create", (done) => {
+      articles.events.on("create", () => done());
+      articles.create({"title": "win"});
+    });
```
Comment (glasserc) [inline test/collection_test.js]: How do these tests work? If you don't call `done()`, does the test fail?

File: test/collection_test.js [hunk 5]
```diff
@@ -2447,4 +2447,160 @@ describe("Collection", () => {
...
+
+    it("should emit an event on deleteAny", (done) => {
+      articles.events.on("delete", () => done());
+      articles.deleteAny(article.id);
+    });
+
+    it("should not emit if deleteAny fails", (done) => {
+      articles.events.on("delete", () => done(new Error("fail")));
+      articles.deleteAny(uuid4());
+      setTimeout(() => done(), 5);
```
Comment (glasserc) [inline test/collection_test.js]: Instead of `setTimeout`, can you just use `.then()` ?

File: docs/api.md (hunk 2 above)
Comment (glasserc) [inline docs/api.md]: OK, I see that there's no special field for `id`, but you can retrieve them from the records provided as `data` and `oldRecord`.

File: test/collection_test.js (hunk 4 above)
Comment (glasserc) [inline test/collection_test.js]: I see. If you don't call `done()`, the test fails after a timeout of 2 sec.

**Comment by glasserc** [comment]
This doesn't even require changes to `fx-src` because an event emitter has been present for almost a year now. r+ with nits. Great work!

**Pull request merged**